### PR TITLE
 Update README and versions for 23.12 branch

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,182 +28,4 @@
 
 [![License](https://img.shields.io/badge/License-BSD3-lightgrey.svg)](https://opensource.org/licenses/BSD-3-Clause)
 
-**LATEST RELEASE: You are currently on the main branch which tracks
-under-development progress towards the next release. The current release branch
-is [r23.10](https://github.com/triton-inference-server/vllm_backend/tree/r23.10)
-and which corresponds to the 23.10 container release on
-[NVIDIA GPU Cloud (NGC)](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tritonserver).**
-
```
Comment (mc-nv) [inline README.md:36]: We need to minimize our impact on `README.md`

Comment (mc-nv) [reply README.md:36]: Let's have it as is, we may wan't to get rid of those headers **LATEST REASE**

Comment (tanmayv25) [reply README.md:36]: Done!

Comment (mc-nv) [reply README.md:36]: That notice is redundant

**Review by mc-nv — APPROVED** [review]

**Pull request merged**